nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -414,7 +415,32 @@ private Description checkParamOverriding(
     @Override
     public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState state) {
         Symbol.MethodSymbol methodSymbol = NullabilityUtil.getFunctionalInterfaceMethod(tree);
-        return checkParamOverriding(tree, tree.getParameters(), methodSymbol, true);
+        Description description = checkParamOverriding(tree, tree.getParameters(), methodSymbol, true);
+        if (description != Description.NO_MATCH) {
+            return description;
+        }
+        if (tree.getBodyKind() == LambdaExpressionTree.BodyKind.EXPRESSION) {
+            ExpressionTree resExpr = (ExpressionTree) tree.getBody();
```

Comment:
Can we extract a method here to share code with `matchReturn`?

Reply:
Sure!